client/dashboard/domains/domain-overview/featured-cards.tsx

```diff
@@ -1,16 +1,20 @@
 import { useSuspenseQuery } from '@tanstack/react-query';
 import { __experimentalGrid as Grid } from '@wordpress/components';
 import { domainQuery } from '../../app/queries/domain';
+import { siteByIdQuery } from '../../app/queries/site';
 import { domainRoute } from '../../app/router/domains';
 import FeaturedCardRenew from './featured-card-renew';
+import FeaturedCardSite from './featured-card-site';
 
 export default function FeaturedCards() {
 	const { domainName } = domainRoute.useParams();
 	const { data: domain } = useSuspenseQuery( domainQuery( domainName ) );
+	const { data: site } = useSuspenseQuery( siteByIdQuery( domain.blog_id ) );
```

Comment:
I think it makes more sense to fetch the site in `<FeaturedCardSite>` instead, so each component is responsible to fetch its own required data. What do you think?

Reply:
That's another way to address the layout shifting too much during load. If `<FeaturedCardSite>` unconditionally renders, then it can show the card border while the load is in progress.

That's how we do it on the site overview page.

Reply:
I moved it to the card component.